src/render.ts

```diff
@@ -0,0 +1,21 @@
+import { type TemplateElement } from './view/abstract-element';
+
+enum RenderPosition {
+  BEFOREBEGIN = 'beforebegin',
+  AFTERBEGIN = 'afterbegin',
+  BEFOREEND = 'beforeend',
+  AFTEREND = 'afterend',
+}
+
+function createElement(template: string): Element {
+  const newElement: HTMLDivElement = document.createElement('div');
+  newElement.innerHTML = template;
+
+  return newElement.firstElementChild as Element;
+}
+
+function render(component: TemplateElement, container: Element, place: RenderPosition = RenderPosition.BEFOREEND) {
```

Comment:
Тоже можно юзать класс и даже импортировать его в файл как тип
```suggestion
function render(component: AbstractElement, container: Element, place: RenderPosition = RenderPosition.BEFOREEND) {
```

Хотя по sol**I**d правильнее скорее так
```suggestion
function render(component: Pick<AbstractElement, 'element'>, container: Element, place: RenderPosition = RenderPosition.BEFOREEND) {
```
У метода `insertAdjacentElement` в ТС уже есть тип на первый параметр. Может воспользоваться им и удалить енамку-объёкт RenderPosition. Такие костыли нужны для ванильного JS, а TS нам и нам будет подсказывать и дополнять строку, и ругать если мы передадим неверную позицию
```suggestion
function render(component: Pick<AbstractElement, 'element'>, container: Element, place: InsertPosition = 'beforeend') {
```